src/Protocol/HttpProtocol.php

```diff
@@ -87,18 +88,28 @@ public function __construct(
 
     /**
      * {@inheritdoc}
+     *
+     * @throws HttpSenderException
      */
     public function send(Receiver $receiver, Notification $notification, bool $sandbox): void
     {
         try {
             $this->doSend($receiver, $notification, $sandbox);
-        } catch (SendNotificationException $e) {
+        } catch (HttpSenderException $e) {
             $this->httpSender->close();
```

Comment:
Won't this line still kill the connection?

Reply:
Yes, but we should close the resource of cURL or other because after hard fail we cannot use this resource. This exception throws only on cURL sender if the function `curl_exec` return fails.

Reply:
@morloderex any questions? Or all OK?

> if yes then please remove it if we are manually creating the protocol ourself?

No, the developer cannot create connection manually because this is has been hard for understanding.

The solution in this PR: correct close HTTP connection to apple, because the apple does not close connection after correct fails (device not found as an example).

We add the `close` method to protocol, because in other cases if an application works in the background, we should close the connection after success/fail to send the notification (via RabbitMQ as an example).

Thanks.